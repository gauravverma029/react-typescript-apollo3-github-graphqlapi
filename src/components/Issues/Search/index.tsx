import React from "react";

type Props = {
  handleFilterStates: (props: FiterStateProps) => void;
};

const Search: React.FC<Props> = ({ handleFilterStates }) => {
  const handleInputChange = (event: React.KeyboardEvent<HTMLInputElement>) => {
    event.stopPropagation();
    if (event.key === "Enter") {
      let target = event.target as HTMLInputElement;
      handleFilterStates({ type: "searchString", value: target.value });
    }
  };

  return (
    <div className="p-4">
      <form
        data-testid="form"
        className="relative"
        onSubmit={(e) => e.preventDefault()}
      >
        <svg
          id="svg"
          width="20"
          height="20"
          fill="currentColor"
          className="absolute left-3 top-1/2 transform -translate-y-1/2 text-gray-400"
        >
          <path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
          />
        </svg>
        <input
          type="text"
          placeholder="Search all Issues"
          className="px-4 border-2 rounded-md border-gray-600 focus:outline-none py-2 pl-10 w-full"
          onKeyDown={handleInputChange}
        />
      </form>
    </div>
  );
};
export default Search;
