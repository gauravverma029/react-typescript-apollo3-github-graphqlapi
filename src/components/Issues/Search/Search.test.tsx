import React from "react";
import { cleanup, render, fireEvent } from "@testing-library/react";

import Search from "./index";

afterEach(cleanup);

const handleFilterStates = () => {};

describe("Search Component", () => {
  test("Should Render", async () => {
    const { asFragment } = render(
      <Search handleFilterStates={handleFilterStates} />
    );
    const html = asFragment();
    expect(html).toMatchSnapshot();
  });

  test("Should Contain Filter All", async () => {
    const { getByTestId, getByPlaceholderText } = render(
      <Search handleFilterStates={handleFilterStates} />
    );
    const searchInput = getByPlaceholderText("Search all Issues");
    fireEvent.change(searchInput, { target: { value: "facebook" } });
    fireEvent.keyDown(searchInput, { key: "Enter", code: "Enter" });
    fireEvent.submit(getByTestId("form"));
  });
});
