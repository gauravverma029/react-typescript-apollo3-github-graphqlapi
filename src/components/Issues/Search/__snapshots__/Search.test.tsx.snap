// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component Should Render 1`] = `
<DocumentFragment>
  <div
    class="p-4"
  >
    <form
      class="relative"
      data-testid="form"
    >
      <svg
        class="absolute left-3 top-1/2 transform -translate-y-1/2 text-gray-400"
        fill="currentColor"
        height="20"
        id="svg"
        width="20"
      >
        <path
          clip-rule="evenodd"
          d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
          fill-rule="evenodd"
        />
      </svg>
      <input
        class="px-4 border-2 rounded-md border-gray-600 focus:outline-none py-2 pl-10 w-full"
        placeholder="Search all Issues"
        type="text"
      />
    </form>
  </div>
</DocumentFragment>
`;
